### PR TITLE
adds guard for reading pipeline resource

### PR DESCRIPTION
**Pull request by esotrope**

### CODE REVIEW

File: script/src/main/java/org/opentestsystem/rdw/script/publishing/ArchiveBackedPublishedPipelineRepository.java [hunk 1]
```diff
@@ -98,10 +98,14 @@ public ArchiveBackedPublishedPipelineRepository(
...
+
+        if (!archiveService.exists(path)) {
+            return null;
+        }
+
         final PipelineResource resource = deserialize(
-                archiveService.readResource(
-                        toPipelinePath(code, version)
-                ),
+                archiveService.readResource(path),
```
Comment (esotrope) [inline script/src/main/java/org/opentestsystem/rdw/script/publishing/ArchiveBackedPublishedPipelineRepository.java]: oops. readResource throws if the resource is absent

File: script/src/main/java/org/opentestsystem/rdw/script/publishing/ArchiveBackedPublishedPipelineRepository.java [hunk 2]
```diff
@@ -98,10 +98,14 @@ public ArchiveBackedPublishedPipelineRepository(
...
                 PipelineResource.class
         );
```
Comment (gregcharles) [inline script/src/main/java/org/opentestsystem/rdw/script/publishing/ArchiveBackedPublishedPipelineRepository.java]: Would this take two calls to the backing archive? Could you just handle the exception and save the extra call?

Comment (esotrope) [reply script/src/main/java/org/opentestsystem/rdw/script/publishing/ArchiveBackedPublishedPipelineRepository.java]: i could. fishing for permission to update the archive service api to return an optional first though ; J

Comment (malaffoon) [reply script/src/main/java/org/opentestsystem/rdw/script/publishing/ArchiveBackedPublishedPipelineRepository.java]: I'm fine with that. You'll need to update all usage of course.

Comment (esotrope) [reply script/src/main/java/org/opentestsystem/rdw/script/publishing/ArchiveBackedPublishedPipelineRepository.java]: 👍 i'll make a jira

**Pull request merged**